Updating Links to HTTPS

bhubbard opened the pull request:
Supporting HTTPS. In order to better support HTTPS, I thought it would be useful to go thru the plugin and see how often HTTP was used. Turns out quite a bit when doing a search. I started going thru the plugin and updating as many links to HTTPS as possible. I attempted to test as much as I could when making the changes, focusing more on links in comments and documentation, but I could use some help with it. 

After my changes I got the `http://` links down to 141. Here is a list of a few links that are common across several files that don't support HTTPS:

* http://en.support.files.wordpress.com
* http://surveys.jetpack.me
* http://ogp.me/ 
* http://sass-lang.com/ 
* http://lesscss.org/ 
* http://sitemaps.org/
* http://afterthedeadline.com
* http://htmldog.com
* http://photomatt.net/ - redirects, but https://photomatt.net gives a security error

Yes, I realize some of these are outside of the control of the jetpack team, although a ping to some of these sites might take it more seriously if the suggestion comes from the jetpack team.

jeherve commented:
This looks good, thanks for the hard @bhubbard!

Before this gets merged, I'd suggesting squashing all these commits into a single one, that would clean things up a bit.

We'll also need to make sure domains like `stats.wordpress.com`, `videopress.com`, or `pixel.wp.com` are optimized for HTTPS. We don't want to slow some of the requests made by Jetpack when switching from http to https.

kraftbj commented on _inc/footer.php:
```diff
@@ -5,20 +5,20 @@
-					<a href="http://wordpress.com/tos/" target="_blank"><?php esc_html_e( 'Terms', 'jetpack' ); ?></a>
-					<a href="http://automattic.com/privacy/" target="_blank"><?php esc_html_e( 'Privacy', 'jetpack' ); ?></a>
+					<a href="https://jetpack.com" target="_blank">Jetpack <?php echo JETPACK__VERSION; ?></a>
+					<a href="https://en.wordpress.com/tos/" target="_blank"><?php esc_html_e( 'Terms', 'jetpack' ); ?></a>
```
For the TOS, we should leave it without the language subdomain. There is code in place to redirect to other languages based on various bits, e.g https://de.wordpress.com/tos/

They will all be https though.

kraftbj commented on class.jetpack-debugger.php:
```diff
@@ -4,7 +4,7 @@ class Jetpack_Debugger {
 	private static function is_jetpack_support_open() {
 		try {
-			$response = wp_remote_request( "http://jetpack.com/is-support-open" );
+			$response = wp_remote_request( "https://jetpack.com/is-support-open" );
```
This brings up an interesting thing we have looked at yet with the broader dotcom move to all mapped domains to https. There are some servers that can pull http, but not https. When https was optional, we intentionally left this as http to allow the most servers possible able to ping us to determine if support was open.

Since last week's switch to all https, we lost this. Something we need to circle back with systems to determine if there's a way to exclude specific URLs from the autoredirect.

kraftbj commented on languages/jetpack-af.po:
```diff
@@ -34,7 +34,7 @@ msgstr ""
 
 #. translators: %s is a URL
 #: class.jetpack.php:1413
-msgid "In <a href=\"%s\" target=\"_blank\">Development Mode</a>, via site URL lacking a dot (e.g. http://localhost)."
```
For the PO files, I'd vote removing them from the PR and let the normal translation process handle the string updates.

kraftbj commented on modules/after-the-deadline/rtl/atd-rtl.css:
```diff
@@ -23,8 +23,8 @@
- * MIT license  : http://www.opensource.org/licenses/mit-license.php
- * Project      : http://jquery-spellchecker.googlecode.com
+ * MIT license  : https://www.opensource.org/licenses/mit-license.php
+ * Project      : https://code.google.com/archive/p/jquery-spellchecker/
```
While we're changing it, it should be https://github.com/badsyntax/jquery-spellchecker now

kraftbj commented on modules/custom-css/csstidy/class.csstidy.php:
```diff
@@ -65,7 +65,7 @@
  * thus has full CSS2 support and a higher reliability.
  * Additional to that it applies some optimisations and fixes to the CSS code.
- * An online version should be available here: http://cdburnerxp.se/cssparse/css_optimiser.php
+ * An online version should be available here: https://cdburnerxp.se/cssparse/css_optimiser.php
```
Let's remove this link. It 404 and the root is for a CD burner.

kraftbj commented on modules/shortcodes/soundcloud.php:
```diff
@@ -1,7 +1,7 @@
 /*
 Plugin Name: SoundCloud Shortcode
-Plugin URI: http://wordpress.org/extend/plugins/soundcloud-shortcode/
+Plugin URI: https://wordpress.org/extend/plugins/soundcloud-shortcode/
```
Might as well remove /extend/ too.

kraftbj commented on modules/shortcodes/wufoo.php:
```diff
@@ -4,7 +4,7 @@
 Author: Chris Coyier / Wufoo, evansolomon
 
-Based on http://wordpress.org/extend/plugins/wufoo-shortcode/
+Based on https://wordpress.org/extend/plugins/wufoo-shortcode/
```
Might as well remove `extend`

kraftbj commented on modules/shortlinks.php:
```diff
@@ -52,7 +52,7 @@ function wpme_get_shortlink( $id = 0, $context = 'post', $allow_slugs = true ) {
 			$id = $blog_id;
 
-		$wpme_url = 'http://wp.me/' . wpme_dec2sixtwo( $id );
+		$wpme_url = 'https://wp.me/' . wpme_dec2sixtwo( $id );
```
We can't do this. wp.me will redirect based on the protocol and this makes the assumptions that all sites using it will be https.

jeherve replied:
I think it's safe. wp.me always redirects to http anyway (see #1498)

kraftbj commented on modules/shortlinks.php:
```diff
@@ -81,7 +81,7 @@ function wpme_get_shortlink( $id = 0, $context = 'post', $allow_slugs = true ) {
 		return '';
 
-	$url = 'http://wp.me/' . $type . wpme_dec2sixtwo( $blog_id ) . '-' . $id;
+	$url = 'https://wp.me/' . $type . wpme_dec2sixtwo( $blog_id ) . '-' . $id;
```
We can't do this. wp.me will redirect based on the protocol and this makes the assumptions that all sites using it will be https.

jeherve replied:
I think it's safe. wp.me always redirects to http anyway (see #1498)

kraftbj commented on modules/theme-tools/random-redirect.php:
```diff
@@ -1,16 +1,16 @@
 /*
 Plugin Name: Random Redirect
-Plugin URI: http://wordpress.org/extend/plugins/random-redirect/
+Plugin URI: https://wordpress.org/extend/plugins/random-redirect/
```
-`/extend`

kraftbj commented on modules/widgets/gravatar-profile.php:
```diff
@@ -230,7 +230,7 @@ function enqueue_scripts() {
 		wp_enqueue_style(
 			'gravatar-card-services',
-			is_ssl() ? 'https://secure.gravatar.com/css/services.css' : 'http://s.gravatar.com/css/services.css',
+			is_ssl() ? 'https://secure.gravatar.com/css/services.css' : 'https://s.gravatar.com/css/services.css',
```
Either need to leave this as a `is_ssl` conditional with the http option or just use one URL. both `secure` and `s` appear to work fine under HTTPS.

kraftbj commented:
I made an initial review, but with so many LOC, I don't feel confident that I hit all of them in my first pass. Leaving it in need of review for additional passes.

bhubbard commented:
Ok, 

I squashed most of the commits per @jeherve suggestions. Looks like one of the builds failed right before, not sure how to fix that one yet: https://travis-ci.org/Automattic/jetpack/jobs/122310818

@kraftbj 
I can revert all the PO files back to before the commits as you suggested. What about the php unit tests? Only the one is failing at the moment, but then the tests did change when I ran thru and updated all the urls.

kraftbj commented:
@bhubbard I'm rerunning the one test. I think it's a false failure that crops up every so often, but we haven't dug in to figure out why it randomly fails.

bhubbard commented:
Closing this pull request, moving it to #3704.
